src/averageAges.js

```diff
@@ -55,10 +62,14 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(({ mother, sex }) =>
+    mother && ((onlyWithSon && sex === 'm') || (!onlyWithSon))
+      && people.find(mom => mother === mom.name));
+
+  const ageDiff = children.map(({ mother: childMother, born }) =>
+    born - people.find(({ name }) => childMother === name).born);
+
+  return ageDiff.reduce((total, value) => total + value, 0) / ageDiff.length;
```

Comment:
You should use the 'getAverageAge' function here instead of repeating the logic to calculate average.